front50-sql/src/test/kotlin/com/netflix/spinnaker/front50/model/SqlStorageServiceTests.kt

```diff
@@ -23,235 +23,253 @@ import com.netflix.spinnaker.front50.model.application.Application
-        ).isEqualTo("id-pipeline001")
-
-        // delete the pipeline
-        sqlStorageService.deleteObject(ObjectType.PIPELINE, "id-pipeline001")
 
-        expectThrows<NotFoundException> {
-          sqlStorageService.loadObject<Pipeline>(ObjectType.PIPELINE, "id-pipeline001")
+          application = sqlStorageService.loadObject(ObjectType.APPLICATION, "application001")
+          expectThat(application.description).isEqualTo("my updated application!")
         }
       }
 
-      test("bulk load pipelines") {
-        val objectKeys = mutableSetOf<String>()
-        (1..10).forEach {
-          val objectKey = "id-pipeline00$it"
-          objectKeys.add(objectKey)
-
+      context("Pipeline") {
+        test("create, update and delete a pipeline") {
+          // verify that a pipeline can be created
           sqlStorageService.storeObject(
             ObjectType.PIPELINE,
-            objectKey,
+            "id-pipeline001",
             Pipeline().apply {
-              id = objectKey
-              name = "pipeline00$it"
+              name = "pipeline001"
               lastModified = 100
 
               put("application", "application001")
             }
           )
-        }
 
-        val pipelines = sqlStorageService.loadObjects<Pipeline>(
-          ObjectType.PIPELINE,
-          (objectKeys + "does_not_exist").toList()
-        )
-        expectThat(
-          pipelines.map { it.id }.toSet()
-        ).isEqualTo(objectKeys)
-      }
-    }
+          var pipeline = sqlStorageService.loadObject<Pipeline>(ObjectType.PIPELINE, "id-pipeline001")
+          expectThat(pipeline.name).isEqualTo("pipeline001")
 
-    context("Entity Tags") {
-      test("create, update and delete an entity tag") {
-        // verify that entity tags can be created
-        sqlStorageService.storeObject(
-          ObjectType.ENTITY_TAGS,
-          "id-entitytags001",
-          EntityTags().apply {
-            id = "id-entitytags001"
-            lastModified = 100
-
-            entityRef = EntityTags.EntityRef().apply {
-              entityId = "application001"
+          // verify that a pipeline can be updated
+          sqlStorageService.storeObject(
+            ObjectType.PIPELINE,
+            "id-pipeline001",
+            Pipeline().apply {
+              name = "pipeline001_updated"
+              lastModified = 200
+
+              put("application", "application001")
             }
+          )
+
+          pipeline = sqlStorageService.loadObject(ObjectType.PIPELINE, "id-pipeline001")
+          expectThat(pipeline.name).isEqualTo("pipeline001_updated")
+
+          expectThat(
+            jooq
+              .select(DSL.field("id", String::class.java))
+              .from("pipelines")
+              .where(
+                DSL.field("name", String::class.java).eq("pipeline001_updated")
+              )
+              .fetchOne(DSL.field("id", String::class.java))
+          ).trim().isEqualTo("id-pipeline001")
```

Comment:
why is the trim needed here?

Reply:
(was specifically added in this changeset)

Reply:
Was receiving string with length equal to size of the column when I run against a Test container. This is not a problem with the real DB.

Reply:
can you confirm that is also the case with real postgres (vs cockroach)?  testcontainer should be running postgres..

Reply:
Good callout.  Postgres TestContainer  and actual postgres are behaving the same way.  Liquibase is converting char datatype to bpchar (blankpadded char) and comparisons are not working when using text of arbitrary length.  Postgres preferred datatype is VARCHAR instead of CHAR ,I will adjust liquibase to use the new type.